Sources/Basics/AuthorizationProvider.swift

```diff
@@ -133,17 +153,25 @@ public struct NetrcAuthorizationProvider: AuthorizationProvider {
 // MARK: - Keychain
 
 #if canImport(Security)
-public struct KeychainAuthorizationProvider: AuthorizationProvider {
+public class KeychainAuthorizationProvider: AuthorizationProvider, AuthorizationWriter {
     private let observabilityScope: ObservabilityScope
+    
+    private let cache = ThreadSafeKeyValueStore<String, (user: String, password: String)>()
 
     public init(observabilityScope: ObservabilityScope) {
         self.observabilityScope = observabilityScope
     }
 
-    public func addOrUpdate(for url: URL, user: String, password: String, callback: @escaping (Result<Void, Error>) -> Void) {
+    public func addOrUpdate(for url: URL, user: String, password: String, persist: Bool = true, callback: @escaping (Result<Void, Error>) -> Void) {
         guard let server = url.authenticationID else {
             return callback(.failure(AuthorizationProviderError.invalidURLHost))
         }
+        
+        if !persist {
```

Comment:
is this mainly for testing, or more broadly needed?

Reply:
This is general feature. `login` must test that the credentials work because persisting them.